### DISCUSSION TITLE
Supertype/subtype relationships

### DISCUSSION THREAD

**Discussion by H14** [directus/directus#12850] [Archive]
Hi

I have a supertype/subtype relationship in my data model. I want to find out if this pattern is somehow compatible with Directus.

<img width="470" alt="Schermafbeelding 2022-04-19 om 15 25 45" src="https://user-images.githubusercontent.com/892420/164014200-1b63c38c-49f4-409d-b4ee-1fcdf20c9f77.png">

The idea is that both `Bugs` and `FeatureRequests` are a subtype of the supertype Issues. Everything common for all issue types is stored in the Issues table, everything specific to the subtypes in their respective tables. The primary keys of `Bugs` and `FeatureRequests` are foreign keys to the auto incremented issue_id from `Issues`. This is very neat pattern! Any other type of relation with another table can also be shared via the common supertype table, like `Category` in this example. And it is super easy to query.

Can I use this pattern within Directus? Or what would be needed to make this or something equivalent to this work with Directus?

Many thanks!

**Comment by rijkvanzanten** [comment]
This pattern is currently not supported in Directus. The API prevents you from creating a m2o field on top of a primary key field, as that has caused many accidents in the past. That being said, it might already work as expected when setting up the tables directly in the database, and have Directus introspect it from there 👍🏻

**Comment by H14** [comment]
Ok, I tried introspecting, that kinda works (but not on everything, sometimes a random error when Directus tries to add another primary key on top of an existing). But the bigger conclusion is that it makes little sense to use the app to manage data modelled this way (as you need to create the issue first and then the bug/feature request manually on top of it). This asks for custom coding. Or reworking the data model to make it make sense with Directus.

Thanks!
